AutofillCredentialProvider/CredentialProvider/CredentialProviderViewController.swift

```diff
@@ -21,56 +21,121 @@ import AuthenticationServices
 import SwiftUI
 import BrowserServicesKit
 import Core
+import Common
+
+class CredentialProviderViewController: ASCredentialProviderViewController {
 
-final class CredentialProviderViewController: ASCredentialProviderViewController {
-    
     private struct Constants {
         static let openPasswords = AppDeepLinkSchemes.openPasswords.url
     }
-    
+
+    private lazy var authenticator = UserAuthenticator(reason: UserText.credentialProviderListAuthenticationReason,
+                                                       cancelTitle: UserText.credentialProviderListAuthenticationCancelButton)
+
+    private lazy var credentialIdentityStoreManager: AutofillCredentialIdentityStoreManaging = AutofillCredentialIdentityStoreManager(credentialStore: ASCredentialIdentityStore.shared,
+                                                                                                                                      vault: secureVault,
+                                                                                                                                      tld: tld)
+
     private lazy var secureVault: (any AutofillSecureVault)? = {
-        return try? AutofillSecureVaultFactory.makeVault(reporter: SecureVaultReporter())
+        try? AutofillSecureVaultFactory.makeVault(reporter: SecureVaultReporter())
     }()
-    
-    private lazy var passwordFetcher: CredentialFetcher = {
-        return CredentialFetcher(secureVault: secureVault)
-    }()
-    
+
+    private lazy var tld: TLD = TLD()
+
+    private lazy var vaultCredentialManager: VaultCredentialManaging = VaultCredentialManager(secureVault: secureVault,
+                                                                                              credentialIdentityStoreManager: credentialIdentityStoreManager)
+
+    // MARK: - ASCredentialProviderViewController Overrides
+
+    override func prepareCredentialList(for serviceIdentifiers: [ASCredentialServiceIdentifier]) {
+        loadCredentialsList(for: serviceIdentifiers)
+    }
+
+    override func provideCredentialWithoutUserInteraction(for credentialIdentity: ASPasswordCredentialIdentity) {
+        // A quirk here is calling .canAuthenticate in this one scenario actually triggers the prompt to authentication
+        // Calling .authenticate here results in the extension attempting to present a non-existent view controller causing weird UI
+        if authenticator.canAuthenticateViaBiometrics() {
```

Comment:
If biometrics authentication is not possible (or no device auth is set), `userInteractionRequired` is returned here. This triggers a call to `prepareInterfaceToProvideCredential(for credentialRequest: any ASCredentialRequest)` which presents a lock screen UI and complete authentication & error handling